libcontainer/cgroups/fs/cpuset.go

```diff
@@ -41,7 +44,128 @@ func (s *CpusetGroup) Set(path string, cgroup *configs.Cgroup) error {
+				extracted = append(extracted, uint16(i))
+			}
+		case 1:
+			value, err := strconv.ParseUint(s, 10, 16)
+			if err != nil {
+				return extracted, err
+			}
+			extracted = append(extracted, uint16(value))
+		}
+	}
+
+	return extracted, nil
+}
+
 func (s *CpusetGroup) GetStats(path string, stats *cgroups.Stats) error {
+
+	var errStrings []string
+
+	if cpus, err := getCpusetStat(path, "cpuset.cpus"); err != nil {
+		errStrings = append(errStrings, err.Error())
+	} else {
+		stats.CpusetStats.Cpus = cpus
+	}
+	if cpuExclusive, err := fscommon.GetCgroupParamUint(path, "cpuset.cpu_exclusive"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.CpuExclusive = cpuExclusive
+	}
+	if mems, err := getCpusetStat(path, "cpuset.mems"); err != nil {
+		errStrings = append(errStrings, err.Error())
+	} else {
+		stats.CpusetStats.Mems = mems
+	}
+	if memHardwall, err := fscommon.GetCgroupParamUint(path, "cpuset.mem_hardwall"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.MemHardwall = memHardwall
+	}
+	if memExclusive, err := fscommon.GetCgroupParamUint(path, "cpuset.mem_exclusive"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.MemExclusive = memExclusive
+	}
+	if memoryMigrate, err := fscommon.GetCgroupParamUint(path, "cpuset.memory_migrate"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.MemoryMigrate = memoryMigrate
+	}
+	if memorySpreadPage, err := fscommon.GetCgroupParamUint(path, "cpuset.memory_spread_page"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.MemorySpreadPage = memorySpreadPage
+	}
+	if memorySpreadSlab, err := fscommon.GetCgroupParamUint(path, "cpuset.memory_spread_slab"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.MemorySpreadSlab = memorySpreadSlab
+	}
+	if memoryPressure, err := fscommon.GetCgroupParamUint(path, "cpuset.memory_pressure"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.MemoryPressure = memoryPressure
+	}
+	if schedLoadBalance, err := fscommon.GetCgroupParamUint(path, "cpuset.sched_load_balance"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.SchedLoadBalance = schedLoadBalance
+	}
+	if schedRelaxDomainLevel, err := fscommon.GetCgroupParamInt(path, "cpuset.sched_relax_domain_level"); err != nil {
+		if !os.IsNotExist(err) {
+			errStrings = append(errStrings, err.Error())
+		}
+	} else {
+		stats.CpusetStats.SchedRelaxDomainLevel = schedRelaxDomainLevel
+	}
```

Comment:
Wondering if something like this would be somewhat cleaner (removing the intermediate variable, and nested `if`;

```suggestion
	stats.CpusetStats.SchedRelaxDomainLevel, err = fscommon.GetCgroupParamInt(path, "cpuset.sched_relax_domain_level")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
```

The whole function would then look something like;

```go
func (s *CpusetGroup) GetStats(path string, stats *cgroups.Stats) error {
	var (
		err        error
		errStrings []string
	)

	stats.CpusetStats.Cpus, err = getCpusetStat(path, "cpuset.cpus")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.CpuExclusive, err = fscommon.GetCgroupParamUint(path, "cpuset.cpu_exclusive")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.Mems, err = getCpusetStat(path, "cpuset.mems")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.MemHardwall, err = fscommon.GetCgroupParamUint(path, "cpuset.mem_hardwall")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.MemExclusive, err = fscommon.GetCgroupParamUint(path, "cpuset.mem_exclusive")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.MemoryMigrate, err = fscommon.GetCgroupParamUint(path, "cpuset.memory_migrate")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.MemorySpreadPage, err = fscommon.GetCgroupParamUint(path, "cpuset.memory_spread_page")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.MemorySpreadSlab, err = fscommon.GetCgroupParamUint(path, "cpuset.memory_spread_slab")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.MemoryPressure, err = fscommon.GetCgroupParamUint(path, "cpuset.memory_pressure")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.SchedLoadBalance, err = fscommon.GetCgroupParamUint(path, "cpuset.sched_load_balance")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}
	stats.CpusetStats.SchedRelaxDomainLevel, err = fscommon.GetCgroupParamInt(path, "cpuset.sched_relax_domain_level")
	if err != nil && !errors.Is(err, os.ErrNotExist) {
		errStrings = append(errStrings, err.Error())
	}

	if len(errStrings) != 0 {
		return fmt.Errorf(strings.Join(errStrings, "\n"))
	}
	return nil
}
```